### PR TITLE
Commentbuttons

### LINKED ISSUE #34: Edits comments on posts
As a social networker
so i can update comments within a specific time range
i want to be able to edit comments within 10 minutes and know the edit has been successful

### LINKED ISSUE #35: Deleting comments on posts
As a social networker
so i can remove drunken comments
i want to be able to delete comments and know that the deletion has been successful

### PR DESCRIPTION
**Pull request by runticle**
fixes #34 
fixes #35

### CODE REVIEW

**Comment by runticle** [comment]
please suggest any refactors! 

Dont ask us to explain it !

**Comment by rsmith88** [comment]
We got waterfall-deletions to work too.

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ def new
 
   def create
     @comment = Comment.create(post_comment_params.merge({user_id: current_user.id, post_id: params[:post_id]}))
+    flash[:danger] = "Comment Added. Making everything about you eh?"
```
Comment (imogenmisso) [inline app/controllers/comments_controller.rb]: Haha! 🤣

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-# require 'activesupport'
-
 class PostsController < ApplicationController
   def new
     @post = Post.new
   end
 
   def create
     @post = Post.create(post_params.merge({user_id: current_user.id}))
+    flash[:danger] = "Post Added. Nobody cares.."
```
Comment (imogenmisso) [inline app/controllers/posts_controller.rb]: You guys are so funny!!!

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -31,8 +32,9 @@ def update
     @post = Post.find(params[:id])
     if @post.editable?
       @post.update(message: params[:post][:message])
+      flash[:danger] = "Post updated. Yet another typo?"
     else
-      flash[:danger] = "fuck off, this is not yours!"
+      flash[:danger] = "Fuck off, this is not yours!"
```
Comment (imogenmisso) [inline app/controllers/posts_controller.rb]: Nice capitalisation!

File: app/models/comment.rb [hunk 4]
```diff
@@ -2,4 +2,9 @@ class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
   validates :comment, presence: true, length: { maximum: 255 }
+
+  def editable?
+    DateTime.now < (self.created_at + 10.minutes)
+  end
+
```
Comment (imogenmisso) [inline app/models/comment.rb]: Nice use of copy and paste!

File: config/routes.rb [hunk 5]
```diff
@@ -12,8 +12,8 @@
   get '/logout', to: 'sessions#destroy'
 
   get 'posts/:id/edit', to: 'posts#edit'
-  
-  get 'comments/:id/edit', to: 'comments#edit'
+
+  get 'posts/:id/comments/:id/edit', to: 'comments#edit'
```
Comment (imogenmisso) [inline config/routes.rb]: 🌟

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -17,13 +18,14 @@ def destroy
   end
 
   def edit
-    @post = Comment.find(params[:id])
+    @comment = Comment.find(params[:id])
+    @post = Post.find(@comment.post_id)
```
Comment (toddpla) [inline app/controllers/comments_controller.rb]:
```suggestion
    
```
this line is not needed - in the view you can access the post be @comment.post

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -19,7 +17,10 @@ def index
 
   def destroy
     @post = Post.find(params[:id])
+    @comments = Comment.where(post: @post)
+    @comments.each { |comment| comment.destroy }
```
Comment (toddpla) [inline app/controllers/posts_controller.rb]: deletion of dependents can be dealt with by this:
`has_many :comments, dependent: :delete_all`

File: app/views/comments/edit.html.erb [hunk 8]
```diff
@@ -1,6 +1,6 @@
-<%= form_for @comment do |form| %>
-  <%= form.label :message %>
-  <%= form.text_area :message %>
+<%= form_for(@comment, url: post_comment_path(@post, @comment)) do |form| %>
```
Comment (toddpla) [inline app/views/comments/edit.html.erb]:
```suggestion
<%= form_for(@comment, url: post_comment_path(@comment.post, @comment)) do |form| %>
```

File: config/routes.rb [hunk 9]
```diff
@@ -12,8 +12,8 @@
   get '/logout', to: 'sessions#destroy'
 
   get 'posts/:id/edit', to: 'posts#edit'
```
Comment (toddpla) [inline config/routes.rb]: not needed

Comment (toddpla) [inline config/routes.rb]: not needed this is dealt with in the resources i believe

**Pull request merged**